utils/dask-setup.sh

```diff
@@ -24,75 +24,141 @@ echo -e "\n"
+    if [[ "0" -lt "$NWORKERS" ]] && [[ "$NWORKERS" -le "$NUM_GPUS" ]]; then
+
+        if [[ "$WHOAMI" = "MASTER" ]]; then
+
+            echo "initializing dask scheduler..."
+            screen -dmS dask_scheduler bash -c "source activate $ENVNAME && dask-scheduler"
+            sleep 5
+            echo "... scheduler started"
+
+        fi
+
+        echo -e "\n"
+        echo "starting $NWORKERS worker(s)..."
+
+        for worker_id in $(seq 1 $NWORKERS)
+        do
+            start=$(( worker_id - 1 ))
+            end=$(( NWORKERS - 1 ))
+            other=$(( start - 1 ))
+            devs=$(seq --separator=, $start $end)
+            second=$(seq --separator=, 0 $other)
+            if [[ "$second" != "" ]]; then
+                devs="$devs,$second"
+            fi
+
+            echo "... starting gpu worker $worker_id"
+            
+            if [[ "$LOG" = "DEBUG" ]]; then
+                export NCCL_DEBUG=WARN
+                export create_worker="source activate $ENVNAME && \
+                                      cuda-memcheck dask-worker $MASTER_IPADDR:$DASK_SCHED_PORT \
+                                                                --host=${MY_IPADDR[0]} --no-nanny \
+                                                                --nprocs=1 --nthreads=1 \
+                                                                --memory-limit=0 --name ${MY_IPADDR[0]}_gpu_$worker_id \
+                                                                --local-directory $DASK_LOCAL_DIR/$name"
+                export logfile="${DASK_LOCAL_DIR}/${MY_IPADDR[0]}_gpu_${worker_id}_log.txt"
+                env CUDA_VISIBLE_DEVICES=$devs screen -dmS gpu_worker_$worker_id \
+                                                           bash -c 'script --flush --command "$create_worker" "$logfile"'
+            elif [[ "$LOG" = "INFO" ]]; then
+                export NCCL_DEBUG=INFO
+                export create_worker="source activate $ENVNAME && \
+                                      dask-worker $MASTER_IPADDR:$DASK_SCHED_PORT \
+                                                  --host=${MY_IPADDR[0]} --no-nanny \
+                                                  --nprocs=1 --nthreads=1 \
+                                                  --memory-limit=0 --name ${MY_IPADDR[0]}_gpu_$worker_id \
+                                                  --local-directory $DASK_LOCAL_DIR/$name"
+                export logfile="${DASK_LOCAL_DIR}/${MY_IPADDR[0]}_gpu_${worker_id}_log.txt"
+                env CUDA_VISIBLE_DEVICES=$devs screen -dmS gpu_worker_$worker_id \
+                                                           bash -c 'script --flush --command "$create_worker" "$logfile"'
+            else
+                export create_worker="source activate $ENVNAME && \
+                                      dask-worker $MASTER_IPADDR:$DASK_SCHED_PORT \
+                                                  --host=${MY_IPADDR[0]} --no-nanny \
```

Comment:
We shouldn't set `--no-nanny` if we aren't profiling or debugging code.

Reply:
Our support for `--no-nanny` on GPUs is quite limited (e.g.) if something goes wrong, it will go wrong again, and again, ..., and debugging it won't be faster. I would say if a worker fails, it shouldn't try to restart.